qe/drivers.py

```diff
@@ -180,9 +209,12 @@ def gen_all_connected_determinant(self, psi_det: Psi_det) -> Psi_det:
                 if any(Excitation.is_connected(det_connected, d) for d in psi_det[:i]):
                     continue
 
-                l.append(det_connected)
+                l_global.append(det_connected)
+
+        global_size = len(l_global)
```

Comment:
Can you refractor the code so we don't need `global_size`? We cannot compute it in the general case